pkg/operator/controller/controller.go

```diff
@@ -326,6 +313,76 @@ func (r *reconciler) ensureInternalRouterServiceForIngress(deployment *appsv1.De
 	return nil
 }
 
+// ensureRouterNamespaceAsset ensures that the router namespace exists and
+// matches the expected namespace.
+func (r *reconciler) ensureRouterNamespaceAsset() error {
+	ns, err := r.ManifestFactory.RouterNamespace()
+	if err != nil {
+		return fmt.Errorf("failed to build router namespace: %v", err)
+	}
+
+	// Need to clone namespace as `ensureRouterAsset` will create a new
+	// router asset if it doesn't exist but if the asset already exists, it
+	// will return us the modified asset (current version from api service).
+	current := ns.DeepCopy()
+	key := types.NamespacedName{Name: ns.Name}
+
+	if _, err := r.ensureRouterAsset(key, current); err != nil {
+		return err
+	}
+
+	// TODO: should we check for all label/annotation changes or just
+	//       specific ones node-selector/run-level?
+	modified := !reflect.DeepEqual(current.Labels, ns.Labels) || !reflect.DeepEqual(current.Annotations, ns.Annotations)
+	modified = modified || !reflect.DeepEqual(current.Spec, ns.Spec)
```

Comment:
The only field in `NamespaceSpec` is `Finalizers`; does the operator really need to re-act if someone futzes with the namespace's finalizers?

Reply:
this comment made me wonder if we should actually ignore annotations we don't specifically care about; no solid use case in my head at the moment, just a quick thought before I forget to bring it up later

Reply:
So that does raise a related question - should we care if say someone adds a label or annotation?  

Meaning in the specific case of the router namespace should we just check that the `openshift.io/run-level` label and `openshift.io/node-selector` match and ignore all other changes?